docker/legacy/Dockerfile

```diff
@@ -0,0 +1,177 @@
+# syntax=docker/dockerfile:1.3
```

Comment:
Commit message nits:

- "to new multi stage Dockerfile" lacks a "to *the* new" or "to *a* new".
- please don't emphasize "multi stage"; that's confusing because the legacy `Dockerfile` also is multi-stage already.

I propose: "This is a preparation for allowing ORT users to stick to the current Dockerfile instead of having to directly upgrade to the upcoming rewritten Dockerfile."